tensorflow/python/ops/array_ops.py

```diff
@@ -1323,12 +1338,13 @@ def _apply_mask_1d(reshaped_tensor, mask, axis=None):
           " are None.  E.g. shape=[None] is ok, but shape=None is not.")
     axis = 0 if axis is None else axis
     shape_tensor[axis:axis + ndims_mask].assert_is_compatible_with(shape_mask)
-
-    leading_size = gen_math_ops.prod(shape(tensor)[axis:axis + ndims_mask], [0])
+    # Musikisomorphie: change shape(tensor) to shape_tensor,
```

Comment:
I don't understand this comment

Reply:
@alextp,  the comment means here the codes used to be:
    leading_size = gen_math_ops.prod(**shape(tensor)**[axis:axis + ndims_mask], [0])
    tensor = reshape(tensor,
                     concat([
                         **shape(tensor)**[:axis], [leading_size],
                         **shape(tensor)**[axis + ndims_mask:]
                     ], 0))

I changed those shape(tensor) to shape_tensor, because otherwise unexpected errors will occurr when _slice_helper() calls boolean_mask().

What do you mean "add a test"?

Reply:
Right; the comment should make no reference to the earlier version of the code since it no longer exists, after you've made your changes :-) 

By "add a test" I mean you can add a unit test to tensorflow/python/kernel_tests/array_ops_test.py . This test should verify that passing a boolean ndarray and a boolean tensor will work.

Reply:
Okay, i will add the corresponding test unit to verify dypes of ndarray and tensor, as well as other necessary cases. Also I will try to rebase this PR.  It is almost 10 p.m at my place, I think I have to keep working on it tomorrow. Again, I will let you know once I am done. Have a nice day.